pkg/scheduler/framework/interface.go

```diff
@@ -184,28 +184,35 @@ func (p PluginToStatus) Merge() *Status {
 	}
 
 	finalStatus := NewStatus(Success)
-	var hasUnschedulableAndUnresolvable, hasUnschedulable bool
 	for _, s := range p {
-		if s.Code() == Error {
-			finalStatus.err = s.AsError()
-		} else if s.Code() == UnschedulableAndUnresolvable {
-			hasUnschedulableAndUnresolvable = true
-		} else if s.Code() == Unschedulable {
-			hasUnschedulable = true
+		switch finalStatus.code {
```

Comment:
I think the most readable approach is to map statuses to precedence numbers. And then replace finalStatus.code if the precedence of s.Status is higher.

Reply:
yes, that's better.